packages/openapi-generator/src/wrapper-files/create-file.ts

```diff
@@ -17,10 +17,15 @@ export async function createFile(
   content: string,
   overwrite: boolean
 ): Promise<void> {
-  return writeFile(join(directoryPath, fileName), wrapContent(content), {
-    encoding: 'utf8',
-    flag: overwrite ? 'w' : 'wx'
-  });
+  const addHeader = !fileName.endsWith('.json');
```

Comment:
I have similar [fix](https://github.com/SAP/cloud-sdk-js/pull/918).

Reply:
we take your way and let the developer decide if the header should be added or not.